api/src/main/java/org/openmrs/module/appointments/service/impl/AppointmentsServiceImpl.java

```diff
@@ -52,8 +61,28 @@ public void setAppointmentAuditDao(AppointmentAuditDao appointmentAuditDao) {
         this.appointmentAuditDao = appointmentAuditDao;
     }
 
+    private boolean validateIfUserHasSelfOrAllAppointmentsAccess(Appointment appointment) {
+        return Context.hasPrivilege(MANAGE_APPOINTMENTS) ||
+                isAppointmentForNoProvider(appointment) ||
+                isCurrentUserSamePersonAsOneOfTheAppointmentProviders(appointment.getProviders());
+    }
+
+    private boolean isAppointmentForNoProvider(Appointment appointment) {
+        return isNull(appointment.getProviders()) || appointment.getProviders().size() == EMPTY_SET_SIZE;
+    }
+
+    private boolean isCurrentUserSamePersonAsOneOfTheAppointmentProviders(Set<AppointmentProvider> providers) {
```

Comment:
IMO, function name can be simplified. like - "IsCurrentUserOneAmongstProviders(appointment)